web/src/echo.ts

```diff
@@ -118,20 +121,22 @@ function hide_retry_spinner($row: JQuery): boolean {
     return true;
 }
 
-function show_message_failed(message_id: number, failed_msg: string): void {
+function show_message_failed(message_id: number, _failed_msg: string): void {
     // Failed to send message, so display inline retry/cancel
     message_live_update.update_message_in_all_views(message_id, ($row) => {
         $row.find(".slow-send-spinner").addClass("hidden");
-        const $failed_div = $row.find(".message_failed");
-        $failed_div.toggleClass("hide", false);
-        $failed_div.find(".failed_text").attr("title", failed_msg);
+        const $message_controls = $row.find(".message_controls");
+        $message_controls.html(render_message_controls_failed_msg());
     });
+    // TODO: Show the `_failed_msg` in the UI, describing the reason for the failure.
```

Comment:
We used to display the `failed_message` string, which contained the reason as to why the message failed to be sent. This was removed in 79b4ac815556b8db2f2516e004f522e084780013. I'm still keeping the argument in the args list with the intent of using it. Will start a #frontend discussion on it soon.

Reply:
Yeah, I think it'd be helpful to display it. Probably a good #design discussion; we may want to rework how we present failures to send a message a bit more broadly.